app/uk/gov/hmrc/pdsauthcheckerapi/controllers/AuthorisationController.scala

```diff
@@ -16,35 +16,177 @@
 
 package uk.gov.hmrc.pdsauthcheckerapi.controllers
 
-import javax.inject._
+import cats.data.{NonEmptyList, ValidatedNel}
+import cats.syntax.apply._
+import cats.syntax.traverse._
+import cats.syntax.validated._
 import play.api.Configuration
+
+import javax.inject._
 import play.api.libs.json.Json
-import play.api.mvc._
-import uk.gov.hmrc.pdsauthcheckerapi.models
-import uk.gov.hmrc.pdsauthcheckerapi.models.PdsAuthRequest
+import play.api.mvc.{Action, ControllerComponents, Request, Result}
+import uk.gov.hmrc.pdsauthcheckerapi.controllers.AuthorisationController.{
+  validateDate,
+  validateEori
+}
+import uk.gov.hmrc.pdsauthcheckerapi.models.{
+  AuthorisedBadRequestCode,
+  AuthorisedBadRequestResponse,
+  DateValidationError,
+  Eori,
+  EoriValidationError,
+  PdsAuthRequest,
+  UnvalidatedRequest,
+  ValidationError
+}
 import uk.gov.hmrc.pdsauthcheckerapi.services.PdsService
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
 
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
 
 @Singleton
-class AuthorisationController @Inject()(cc: ControllerComponents, config: Configuration, pdsService: PdsService)(implicit ec: ExecutionContext) extends BackendController(cc)  {
+class AuthorisationController @Inject() (
+    cc: ControllerComponents,
+    config: Configuration,
+    pdsService: PdsService
+)(implicit ec: ExecutionContext)
+    extends BackendController(cc) {
+
+  def authorise: Action[UnvalidatedRequest] =
+    Action(parse.json[UnvalidatedRequest]).async {
+      implicit request: Request[UnvalidatedRequest] =>
+        if (!supportedAuthTypes.contains(request.body.authType)) {
+          Future.successful(InvalidAuthTypeResponse)
+        } else {
 
-  private val supportedAuthTypes: Set[String] = config.get[String]("auth.supportedTypes").split(",").map(_.trim).toSet
+          val handleValidationErrors
```

Comment:
This controller isn't workable any more. Validation should be refactored into it's own service (like we discussed in the prototype) with some sort of Either we can handle.

I do think we have a situation that justifies a ConverterService as well though, to create our response models from our internal models (successful or otherwise).

This will make this controller much easier to understand.